go/vt/vtgate/executor.go

```diff
@@ -528,9 +528,6 @@ func (e *Executor) handleSet(ctx context.Context, safeSession *SafeSession, sql
 	if err != nil {
 		return &sqltypes.Result{}, vterrors.New(vtrpcpb.Code_INVALID_ARGUMENT, err.Error())
 	}
-	if len(vals) > 0 && charset != "" {
-		return &sqltypes.Result{}, vterrors.New(vtrpcpb.Code_INVALID_ARGUMENT, "unexpected key values and charset, must specify one")
-	}
```

Comment:
This check was there before to catch a charset and var change when they happened in the same statement. Now that is handled fine by the parser, so we can remove the check.
cc @bbeaudreault as you added to initial check in here https://github.com/arthurnn/vitess/commit/5faaac2315f553539bf5f0867e3f639ddea91aff